cli/tests/unit/webcrypto_test.ts

```diff
@@ -608,6 +608,109 @@ Deno.test(async function testAesCbcEncryptDecrypt() {
   assertEquals(new Uint8Array(decrypted), new Uint8Array([1, 2, 3, 4, 5, 6]));
 });
 
+Deno.test(async function testAesCtrEncryptDecrypt() {
+  async function aesCtrRoundTrip(
+    key: CryptoKey,
+    counter: Uint8Array,
+    length: number,
+    plainText: Uint8Array,
+  ) {
+    const cipherText = await crypto.subtle.encrypt(
+      {
+        name: "AES-CTR",
+        counter,
+        length,
+      },
+      key,
+      plainText,
+    );
+
+    assert(cipherText instanceof ArrayBuffer);
+    assertEquals(cipherText.byteLength, plainText.byteLength);
+
+    const decryptedText = await crypto.subtle.decrypt(
+      {
+        name: "AES-CTR",
+        counter,
+        length,
+      },
+      key,
+      cipherText,
+    );
+
+    assert(decryptedText instanceof ArrayBuffer);
+    assertEquals(decryptedText.byteLength, plainText.byteLength);
+    assertEquals(new Uint8Array(decryptedText), plainText);
+  }
+  try {
+    for (const keySize of [128, 192, 256]) {
+      const key = await crypto.subtle.generateKey(
+        { name: "AES-CTR", length: keySize },
+        true,
+        ["encrypt", "decrypt"],
+      ) as CryptoKey;
+
+      // test normal operation
+      for (const length of [128 /*, 64, 128 */]) {
+        const counter = await crypto.getRandomValues(new Uint8Array(16));
+
+        await aesCtrRoundTrip(
+          key,
+          counter,
+          length,
+          new Uint8Array([1, 2, 3, 4, 5, 6]),
+        );
+      }
+
+      // test counter-wrapping
+      for (const length of [32, 64, 128]) {
+        const plaintext1 = await crypto.getRandomValues(new Uint8Array(32));
+        const counter = new Uint8Array(16);
+
+        const ciphertext1 = await crypto.subtle.encrypt(
+          {
+            name: "AES-CTR",
+            counter,
+            length,
+          },
+          key,
+          plaintext1,
+        );
+
+        // Set [length] counter bits to all '1's
+        for (let off = 16 - (length / 8); off < 16; ++off) {
+          counter[off] = 0xff;
+        }
+
+        // = [ 1 block of 0x00 + plaintext1 ]
+        const plaintext2 = new Uint8Array(48);
+        plaintext2.set(plaintext1, 16);
+
+        const ciphertext2 = await crypto.subtle.encrypt(
+          {
+            name: "AES-CTR",
+            counter,
+            length,
+          },
+          key,
+          plaintext2,
+        );
+
+        // If counter wrapped, 2nd block of ciphertext2 should be equal to 1st block of ciphertext1
+        // since ciphertext1 used counter = 0x00...00
+        // and ciphertext2 used counter = 0xFF..FF which should wrap to 0x00..00 without affecting
+        // higher bits
+        assertEquals(
+          new Uint8Array(ciphertext1),
+          new Uint8Array(ciphertext2).slice(16),
+        );
+      }
+    }
+  } catch (E) {
+    console.log("Error: ", E);
+  }
```

Comment:
This try/catch block shouldn't be there, right?

Reply:
sure thing! ...gone